mock/mock.go

```diff
@@ -1006,6 +1007,13 @@ func (args Arguments) Diff(objects []interface{}) (string, int) {
 			switch expected := expected.(type) {
 			case anythingOfTypeArgument:
 				// type checking
+				if actual == nil {
+					if expected != "<nil>" {
+						differences++
+						output = fmt.Sprintf("%s\t%d: FAIL:  type %s != type <nil> - %s\n", output, i, expected, actualFmt)
+					}
+					continue
```

Comment:
I don't get the comparison on `"<nil>"` here.

Can't we check on a real value?

I'm paranoid but won't the code react to a string being `"<nil>"`